### PR TITLE
Duplicate mock

### PR DESCRIPTION
**Pull request by giladfuchs**
#367

### CODE REVIEW

File: client/src/pages/mocks/CreateMock.tsx [hunk 1]
```diff
@@ -132,6 +132,9 @@ export const CreateMock: React.FC<CreateMockProps> = ({
         }}
         handleAddMockResponse={onAddResponse}
         handleDeleteMockResponse={onDeleteResponse}
+        // @ts-ignore
+        handleDuplicateMockResponse={undefined}/* TO PASS TS ERROR
```
Comment (ypotter) [inline client/src/pages/mocks/CreateMock.tsx:136]: You can make `handleDuplicateMockResponse` as optional prop.
Than inside the component I saw that you have handled this scenario.

File: client/src/pages/mocks/EditMock.tsx [hunk 2]
```diff
@@ -122,6 +122,27 @@ export const EditMock: React.FC<EditMockProps> = ({ mock, setMock }) => {
     });
   };
 
+  const onDuplicateMockResponse = (mockToDuplicate: MockResponse) => {
+    const index = mock.mockResponses.length;
+
+    return postMockResponse(snifferId as string, mockId as string, {
+      name: `${mockToDuplicate.name} copy ${index}`,
```
Comment (ypotter) [inline client/src/pages/mocks/EditMock.tsx:129]: I would remove the `${index}`.
The addition of copy is a standard by itself.

File: client/src/pages/mocks/EditMock.tsx [hunk 3]
```diff
@@ -122,6 +122,27 @@ export const EditMock: React.FC<EditMockProps> = ({ mock, setMock }) => {
...
+      body: mockToDuplicate.body,
+      status: mockToDuplicate.status,
+      headers: mockToDuplicate.headers,
+      sequenceIndex: index,
+    }).then((res: MockResponse) => {
+      setMock((prev) => {
+        if (!prev) return prev;
+        return {
+          ...prev,
+          selectedResponseId: index === 0 ? res.id : prev.selectedResponseId,
```
Comment (ypotter) [inline client/src/pages/mocks/EditMock.tsx:139]: This line is unnecessary as the index would never be 0 in this case.

File: client/src/pages/mocks/MockResponseCard.tsx [hunk 4]
```diff
@@ -126,7 +128,18 @@ export const MockResponseCard: React.FC<IMockResponseCard> = ({
             </div>
           </div>
           <div className="flex flex-row items-center space-x-2">
-            {isDeleting ? (
+
+         {   onDuplicateMockResponse&&  <Tooltip
+                title="Duplicate mock response"
+              >
+                <FileCopy sx={{fontSize: '15px' }}
```
Comment (ypotter) [inline client/src/pages/mocks/MockResponseCard.tsx:135]: The copy icon isn't accurate for duplicate.
Try to use one of those:
https://react-icons.github.io/react-icons/search/#q=duplicate

**Review by ypotter — CHANGES_REQUESTED** [review]

File: client/src/pages/mocks/MockResponseCard.tsx (hunk 4 above)
Comment (giladfuchs) [reply client/src/pages/mocks/MockResponseCard.tsx:135]: ok

File: client/src/pages/mocks/EditMock.tsx (hunk 3 above)
Comment (giladfuchs) [reply client/src/pages/mocks/EditMock.tsx:139]: ok

File: client/src/pages/mocks/EditMock.tsx (hunk 2 above)
Comment (giladfuchs) [reply client/src/pages/mocks/EditMock.tsx:129]: ok

File: client/src/pages/mocks/EditMock.tsx (hunk 3 above)
Comment (ypotter) [reply client/src/pages/mocks/EditMock.tsx]: `...prev` destruct selectedResponseId so you don't need this line.

File: client/src/pages/mocks/CreateMock.tsx (hunk 1 above)
Comment (ypotter) [reply client/src/pages/mocks/CreateMock.tsx:136]: What about this?

**Review by ypotter — APPROVED** [review]

**Pull request merged**